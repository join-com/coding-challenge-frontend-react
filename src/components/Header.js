import React from 'react';

const Header = () => {
    return (
        <div>
            <div className="ui inverted vertical masthead center aligned segment">
                <div className="ui text container">
                    <h1 className="ui inverted header">
                        Berlin Police Department
                    </h1>
                    <h2>Stolen Bike Index</h2>
                </div>
            </div>
        </div>
    );
}

export default Header;