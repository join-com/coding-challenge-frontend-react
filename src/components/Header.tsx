import React from 'react'
import { withStyles, createStyles } from '@material-ui/core/styles'

import Typography from '@material-ui/core/Typography'
import Grid from '@material-ui/core/Grid'

import logo from '../logo.svg'

const styles = (theme: any) => createStyles({
  header: {
    fontSize: theme.typography.pxToRem(20),
    marginBottom: 5,
  },
  subHeader: {
    fontSize: theme.typography.pxToRem(18),
  },
  wrapper: {
    marginBottom: 20,
  },
  cell: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
  },
  logo: {
    width: 100,
    height: 'auto',
    maxWidth: '80%',
  },
});

interface HeaderProps {
  classes: {
    wrapper: string
    header: string
    subHeader: string
    logo: string
    cell: string
  }
}

class Header extends React.PureComponent<HeaderProps> {
  render() {
    const { classes } = this.props;
    return (
      <Grid container className={classes.wrapper}>
        <Grid item xs={3} sm={2} md={1} className={classes.cell}>
          <img src={logo} className={classes.logo} alt='' />
        </Grid>
        <Grid item xs={9} sm={10} md={11} className={classes.cell}>
          <Typography className={classes.header}>
            Police department of Berlin
          </Typography>
          <Typography className={classes.subHeader}>
            Stolen bikes
          </Typography>
        </Grid>
      </Grid>
    )
  }
}

export default withStyles(styles)(Header);
