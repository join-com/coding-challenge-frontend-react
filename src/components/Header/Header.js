import React from 'react'

import Logo from './logo.png'

const Header = () => (
  <div>
    <img src={Logo} alt="logo" />
  </div>
)

export default Header
