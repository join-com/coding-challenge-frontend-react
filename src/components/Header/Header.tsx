import React from "react";
import { Typography } from "@material-ui/core";
import { withStyles, WithStyles, createStyles } from "@material-ui/core/styles";

import logo from "./logo.png";

const styles = createStyles({
  header: {
    display: "flex",
    alignItems: "center"
  },
  logo: {
    width: 150,
    height: 150,
    marginRight: 24
  }
});

interface Props extends WithStyles<typeof styles> {}

const Header: React.FunctionComponent<Props> = ({ classes }) => {
  return (
    <header className={classes.header}>
      <a href="/">
        <img src={logo} alt="Logo" className={classes.logo} />
      </a>
      <div>
        <Typography variant="h3" gutterBottom>
          Police Department of Berlin
        </Typography>
        <Typography variant="h5">Stolen Bikes</Typography>
      </div>
    </header>
  );
};

export default withStyles(styles)(Header);
