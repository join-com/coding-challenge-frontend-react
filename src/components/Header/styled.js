import styled from 'styled-components'

const Container = styled.div`
  display: flex;
  justify-content: center;
`

const Logo = styled.img`
  width: 100px;
  height: 100px;
  margin: 0 20px;
`

export { Container, Logo }
