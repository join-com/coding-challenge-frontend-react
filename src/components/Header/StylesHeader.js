import styled from "styled-components";
import { colors, defaultIndent } from "../../constants/styles";

export const StyledHeader = styled.div`
  background-color: ${colors.primary};
  padding: ${defaultIndent * 3}rem 0;
`;
