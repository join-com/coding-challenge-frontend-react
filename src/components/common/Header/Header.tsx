import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';

import { HeaderTextBlock } from 'components';
import { media } from 'ui/utils';
import logo from './assets/logo.png';

const Container = styled(Link)`
    display: flex;
    align-items: center;
	cursor: pointer;

	${media.mobile`
		height: 60px;
		margin-bottom: 15px;
  	`}

	${media.tablet`
    	height: 100px;
	`}

    ${media.desktop`
    	height: 140px;
	`}
`;

const Logo = styled.img`
    width: 64px;
    height: 64px;

    ${media.tabletNDesktop`
		padding: 0 18px;
	`}
`;

export const Header = () => (
    <Container to={'/'}>
        <Logo src={logo} />
        <HeaderTextBlock />
    </Container>
);
