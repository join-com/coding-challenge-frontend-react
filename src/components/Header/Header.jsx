import React from "react";
import { Link } from "react-router-dom";

import { StyledHeader } from "./StylesHeader";
import Heading from "../ui/Heading";
import { Container } from "../ui/Layout/StyledLayout";

const Header = () => {
  return (
    <StyledHeader>
      <Container>
        <Link to="/">
          <Heading>Police Department of Berlin</Heading>
        </Link>
        <Heading level={3}>Stolen bikes</Heading>
      </Container>
    </StyledHeader>
  );
};

export default Header;
