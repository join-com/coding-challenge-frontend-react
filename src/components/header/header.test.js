import React from 'react';
import { shallow } from 'enzyme';
import Header from './header';

describe('MyComponent', () => {
    it('Header should render correctly', () => {
        const component = shallow(<Header />);

        expect(component).toMatchSnapshot();
    });
});