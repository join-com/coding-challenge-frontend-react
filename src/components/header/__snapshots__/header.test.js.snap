// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent Header should render correctly 1`] = `
<Header
  classes={
    Object {
      "logo": "Header-logo-1",
    }
  }
/>
`;
